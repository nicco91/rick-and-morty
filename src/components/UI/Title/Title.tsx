import styled from 'styled-components';

const Title = styled.h1`
  color: ${(props) => props.theme.colors.textPrimary};
  font-size: 64px;
  font-weight: bold;
`;

export default Title;
