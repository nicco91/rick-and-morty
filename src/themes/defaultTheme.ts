import { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#0066cc',
    success: '#00cc66',
    error: '#cc0000',
    textPrimary: '#333',
    textSecondary: '#999',
  },
  breakpoints: {
    mobile: 320,
    mobileLandscape: 480,
    tablet: 768,
    tabletLandscape: 1024,
    desktop: 1200,
    desktopLarge: 1500,
    desktopWide: 1920,
  },
};
