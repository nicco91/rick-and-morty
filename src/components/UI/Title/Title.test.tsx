import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'themes';
import Title from './Title';

it('render a title', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <Title>I am a title!</Title>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
