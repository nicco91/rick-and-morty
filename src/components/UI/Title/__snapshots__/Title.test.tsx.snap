// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a title 1`] = `
.c0 {
  color: #333;
  font-size: 64px;
  font-weight: bold;
}

<h1
  className="c0"
>
  I am a title!
</h1>
`;
